lib/Importer/ONNXModelLoader.cpp

```diff
@@ -1230,6 +1230,541 @@ Error ONNXModelLoader::loadWhere(const ONNX_NAMESPACE::NodeProto &op,
+    RETURN_ERR_IF_NOT(initial_h.dims().size() == 3 &&
+                          initial_h.dims()[0] == numDirections &&
+                          initial_h.dims()[1] == batchSize &&
+                          initial_h.dims()[2] == hiddenSize,
+                      "ONNX LSTM 'initial_h' tensor size invalid!");
+  }
+
+  // Input6: initial_c (Optional).
+  NodeValue initial_c = nullptr;
+  if (numInputs > 6 && !op.input(6).empty()) {
+    ASSIGN_VALUE_OR_RETURN_ERR(initial_c, getNodeValueByName(op.input(6)));
+    RETURN_ERR_IF_NOT(initial_c.dims().size() == 3 &&
+                          initial_c.dims()[0] == numDirections &&
+                          initial_c.dims()[1] == batchSize &&
+                          initial_c.dims()[2] == hiddenSize,
+                      "ONNX LSTM 'initial_c' tensor size invalid!");
+  }
+
+  // Input7: P (Optional).
+  Constant *P = nullptr;
+  if (numInputs > 7 && !op.input(7).empty()) {
+    RETURN_ERR_IF_NOT(hasConstantByName(op.input(7)),
+                      "ONNX LSTM weight 'P' is expected to be constant!");
+    ASSIGN_VALUE_OR_RETURN_ERR(P, getConstantByName(op.input(7)));
+  }
+
+  // -------------------------- Outputs ---------------------------------------
+  // We always create placeholders for the LSTM state variables (Y_h and Y_c)
+  // for the following reasons:
+  // - expose the LSTM state in the graph interface for accessibility (set
+  //   desired state, reset state, watch the state being updated automatically).
+  // - since the LSTM cells are unrolled (no graph loop primitive available
+  //   at this point), the optimal way to use the LSTM within a model would be
+  //   to have it defined with only 1 time step and have the loop in the top
+  //   of the application while the LSTM state will be automatically updated
+  //   from one iteration (time step) to the next through the placeholders.
+
+  const int numOutputs = op.output_size();
+  RETURN_ERR_IF_NOT(1 <= numOutputs,
+                    "ONNX LSTM should have minimum 1 output defined!");
+
+  // Create Y_h (hidden state) output placeholder.
+  Placeholder *Y_h_ph;
+  TypeRef Htype = G_.getParent()->uniqueTypeWithNewShape(
+      X.getType(), {numDirections, batchSize, hiddenSize});
+  std::string Hname = opName + ".Y_h";
+  ASSIGN_VALUE_OR_RETURN_ERR(Y_h_ph,
+                             createAndRegisterPlaceholder(Hname, Htype));
+  inputVarsByName_.try_emplace(Hname, Y_h_ph);
+
+  // Create Y_c (cell state) output placeholder.
+  Placeholder *Y_c_ph;
+  TypeRef Ctype = G_.getParent()->uniqueTypeWithNewShape(
+      X.getType(), {numDirections, batchSize, hiddenSize});
+  std::string Cname = opName + ".Y_c";
+  ASSIGN_VALUE_OR_RETURN_ERR(Y_c_ph,
+                             createAndRegisterPlaceholder(Cname, Ctype));
+  inputVarsByName_.try_emplace(Cname, Y_c_ph);
+
+  // Create X slices.
+  std::vector<Node *> Xslices;
+  if (seqLength == 1) {
```

Comment:
No big deal, but curious if this was required for this to work, or just as an optimization? I think our graph optimizations would fix this later anyway.

Reply:
This is related to optimization: is there is an optimization where a slice node which extract the entire input is washed away by optimization:? If yes, then "if" should dissappear.

Reply:
Thanks! I removed this "manual" optimization then since it will be soon handled by the optimizer.